ios/RCTOneSignal/RCTOneSignal.m

```diff
@@ -42,101 +42,19 @@ + (RCTOneSignal *) sharedInstance {
 }
 
 - (void)initOneSignal {
-    [OneSignal setValue:@"react" forKey:@"mSDKType"];
-    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(didBeginObserving) name:@"didSetBridge" object:nil];
-    
-    [OneSignal initWithLaunchOptions:nil appId:nil handleNotificationReceived:^(OSNotification* notification) {
-        [self handleRemoteNotificationReceived:[notification stringify]];
-    } handleNotificationAction:^(OSNotificationOpenedResult *result) {
-        if (!RCTOneSignal.sharedInstance.didStartObserving)
-            coldStartOSNotificationOpenedResult = result;
-        else
-            [self handleRemoteNotificationOpened:[result stringify]];
-        
-    } settings:@{@"kOSSettingsKeyInOmitNoAppIdLogging" : @true, kOSSettingsKeyAutoPrompt : @false}]; //default autoPrompt to false since init will be called again
-    didInitialize = false;
-}
-
-// deprecated init methods
-// provides backwards compatibility
-- (id)initWithLaunchOptions:(NSDictionary *)launchOptions appId:(NSString *)appId {
-    return [self initWithLaunchOptions:launchOptions appId:appId settings:nil];
-}
-
-- (id)initWithLaunchOptions:(NSDictionary *)launchOptions appId:(NSString *)appId settings:(NSDictionary*)settings {
-    [self configureWithAppId:appId settings:settings];
-    
-    return self;
-}
-
-- (void)didBeginObserving {
-    // To continue supporting deprecated initialization methods (which create a new RCTOneSignal instance),
-    // we will only access the didStartObserving property of the shared instance to avoid issues
-    RCTOneSignal.sharedInstance.didStartObserving = true;
-    
-    dispatch_async(dispatch_get_main_queue(), ^{
-        if (coldStartOSNotificationOpenedResult) {
-            [self handleRemoteNotificationOpened:[coldStartOSNotificationOpenedResult stringify]];
-            coldStartOSNotificationOpenedResult = nil;
-        }
-    });
-}
 
-- (void)dealloc {
-    [[NSNotificationCenter defaultCenter] removeObserver:self];
-}
-
-- (void)configureWithAppId:(NSString *)appId {
-    [self configureWithAppId:appId settings:nil];
-}
-
-- (void)configureWithAppId:(NSString *)appId settings:(NSDictionary*)settings {
-    
     if (didInitialize)
         return;
-    
+
     didInitialize = true;
     [OneSignal addSubscriptionObserver:self];
```

Comment:
In Android we added these observers only when the developer calls them. We should do the same for iOS to keep them consistent.

Reply:
Done.